swarm/network/stream/streamer_test.go

```diff
@@ -921,3 +930,112 @@ func TestMaxPeerServersWithoutUnsubscribe(t *testing.T) {
 		}
 	}
 }
+
+/*
+TestGetSubscriptionsRPC sets up a simulation network of 16 nodes,
+starts the simulation, waits for SyncUpdateDelay in order to kick off
+stream registration, then tests that there are subscriptions.
+If provided with the `-printstats = true` option, it will print
+the information of who is subscribed to who to STDOUT
+*/
+func TestGetSubscriptionsRPC(t *testing.T) {
+	//arbitrarily set to 16
+	nodeCount := 16
+	//set the syncUpdateDelay for sync registrations to start
+	syncUpdateDelay := 500 * time.Millisecond
+	//create a standard sim
+	sim := simulation.New(map[string]simulation.ServiceFunc{
+		"streamer": func(ctx *adapters.ServiceContext, bucket *sync.Map) (s node.Service, cleanup func(), err error) {
+			n := ctx.Config.Node()
+			addr := network.NewAddr(n)
+			store, datadir, err := createTestLocalStorageForID(n.ID(), addr)
+			if err != nil {
+				return nil, nil, err
+			}
+			localStore := store.(*storage.LocalStore)
+			netStore, err := storage.NewNetStore(localStore, nil)
+			if err != nil {
+				return nil, nil, err
+			}
+			kad := network.NewKademlia(addr.Over(), network.NewKadParams())
+			delivery := NewDelivery(kad, netStore)
+			netStore.NewNetFetcherFunc = network.NewFetcherFactory(dummyRequestFromPeers, true).New
+			//configure so that sync registrations actually happen
+			r := NewRegistry(addr.ID(), delivery, netStore, state.NewInmemoryStore(), &RegistryOptions{
+				Retrieval:       RetrievalEnabled,
+				Syncing:         SyncingAutoSubscribe, //enable sync registrations
+				SyncUpdateDelay: syncUpdateDelay,
+			}, nil)
+
+			bucket.Store(bucketKeyRegistry, r)
+			cleanup = func() {
+				os.RemoveAll(datadir)
+				netStore.Close()
+				r.Close()
+			}
+
+			return r, cleanup, nil
+
+		},
+	})
+	defer sim.Close()
+
+	ctx, cancelSimRun := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancelSimRun()
+
+	//upload a snapshot
+	err := sim.UploadSnapshot(fmt.Sprintf("testing/snapshot_%d.json", nodeCount))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	//wait till healthy
+	if _, err := sim.WaitTillHealthy(ctx, 2); err != nil {
+		t.Fatal(err)
+	}
+
+	result := sim.Run(ctx, func(ctx context.Context, sim *simulation.Simulation) error {
+		//we need to wait for some time until registrations are finished...
+		time.Sleep(syncUpdateDelay + 1*time.Second)
```

Comment:
We really should have a better way of telling this.

Reply:
I am all ears if you have a better one. See the comment above https://github.com/ethereum/go-ethereum/pull/18284#discussion_r. 

We need to wait for syncing to kick in and to have some time passed. Otherwise there is nothing to check.

Reply:
We've talked a bit before about how to determine states of protocols when testing. So far it's been inconclusive, though.

That said, aren't all of these syncing things actual protocol messages? Isn't it then more natural to audit the actual messages being exchanged? So when a node actually sends a "streamMeWhateverMessage" that means that node started its syncing process?

Reply:
Addressed by latest version, please review and mark resolved if ok, else let's discuss further